src/Client.php

```diff
@@ -25,295 +26,233 @@ class Client
-        }
-        return $ret;
+        $data = ["member" => $id, "hash" => password_hash($password, PASSWORD_DEFAULT)];
+        var_dump($data);
+        return $this->request("POST", "items/PinHashes", $data);
     }
 
-    public function createPerson($values)
+    /** @brief Get a member list from the server
+     *
+     * TODO: Optionally return active members only
+     */
+    public function getMemberList($active = false) : array
     {
-        $response = $this->guzzleClient->request('POST', 'api/v2/management/person/create', [
-          'form_params' => $values
-        ]);
+        if ($active === true)
+            return $this->getActiveMemberList();
+        $data = $this->request("GET", "items/Members");
+        return $this->mapMembersToArray($data);
+    }
 
-        $arr = json_decode($response->getBody(), true);
-        $this->checkForError($arr);
-        return true;
+    private function today() : string {
+        return date("Y-m-d", time());
     }
 
-    public function updatePerson($id, $values)
-    {
-        $response = $this->guzzleClient->request('POST', 'api/v2/management/person/update', [
-          'form_params' => array_merge([
-            'person_id' => $id
-          ], $values)
+    private function getActiveMemberList() : array {
+        $data = $this->request("SEARCH", "items/Memberships", [
+            "query" => ["filter" => [
+                // "type.end" => ["_gte" => $this->today()]
+                "id" => ["_eq" => 1]
+            ]]
         ]);
-
-        $arr = json_decode($response->getBody(), true);
-        $this->checkForError($arr);
-        return true;
+        var_dump($data);
+        $ids = array();
+        foreach ($data as $membership)
+            $ids[] = $membership["member"];
+        $data = $this->request("GET", "items/Members", ["data" => $ids]);
+        return $this->mapMembersToArray($data);
     }
 
-    public function getMembershipsByID($id)
-    {
-        $response = $this->guzzleClient->request('GET', 'api/v2/model', [
-          'query' => ['model_name' => 'person_model', 'method_name' => "get_valid_memberships_by_person_id", "person_id" => (string) $id]
-      ]);
-        $body = json_decode($response->getBody(), true);
-        $this->checkForError($body);
-        $ret = [];
-        foreach ($body as $membership) {
-            $ret[] = [
-            "active" => $membership['active'],
-            "name" => $membership['name'],
-            "fee" => $membership['fee'],
-            "issue_date" => $membership['issue_date'],
-            "expiry_date" => $membership['expiry_date'],
-            "is_general_membership" => $membership['is_general_membership']
-          ];
+    private function mapMembersToArray($members) : array {
+        $result = array();
+        foreach ($members as $person) {
+            $result[] = [
+                "id" => $person["id"],
+                "active" => true,  // TODO: This is no longer present in Directus
```

Comment:
What do you do with old members in Directus? We always wanted to keep a log of those as well to contact them for Lustrum oid.

Reply:
For now, I've imported them into the new system. Also, I have added an 'active' field, because Directus cannot do queries on joined tables directly in the web-interface, limiting the possible behaviour a bit.